runtime/runc_jailer.go

```diff
@@ -138,10 +138,18 @@ func newRuncJailer(
 
 func (j *runcJailer) prepareBindMounts(mounts []*proto.FirecrackerDriveMount) error {
 	for _, m := range mounts {
-		err := j.bindMountFileToJail(m.HostPath, filepath.Join(j.RootPath(), m.HostPath))
-		if err != nil {
+		stat := syscall.Stat_t{}
+		if err := syscall.Stat(m.HostPath, &stat); err != nil {
 			return err
 		}
+		// Only bindMount regular file.
+		// To avoid duplicate files, for block device, we will use system call to create device file for it.
+		if stat.Mode&syscall.S_IFMT == syscall.S_IFREG {
```

Comment:
Instead of filtering out all non-regular file from bind mount, what about gracefully handling the failure if we encounter the "File exist" error, and move on with the rest of the mount points? The reason being, we are not absolutely sure if we can skip all other devices. I think its safer to just skip the duplicate one and move on to the next one rather than not at all looking at all non-regular files.

Looks like out of these, we are only bind mounting S_IFREG and skipping BLK, CHR, DIR, LNK, FIFO etc. Seems like a problem to me. Correct me if i am wrong. 
Ref: https://www.gnu.org/software/libc/manual/html_node/Testing-File-Type.html

Reply:
FIFO and CHR LNK are all deal in different places. There are special handlers and functions for them as well. The reason why we don't handle the "File exist" error where we encounter it due to (you can read more details on my above discussion)
1. No possible way to workaround without introducing complexity or unnecessary
2. We should remove the blockDevice file at the first place where it being mounted, and here it is.